plugins/catalog-graph/src/alpha.tsx

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import {
+  createPageExtension,
+  createPlugin,
+  createSchemaFromZod,
+} from '@backstage/frontend-plugin-api';
+import {
+  compatWrapper,
+  convertLegacyRouteRef,
+} from '@backstage/core-compat-api';
+import { createEntityCardExtension } from '@backstage/plugin-catalog-react/alpha';
+import { catalogGraphRouteRef, catalogEntityRouteRef } from './routes';
+import { ALL_RELATION_PAIRS, Direction } from './components';
+
+type Zod = Parameters<Parameters<typeof createSchemaFromZod>[0]>[0];
+
+function getCompoundEntityRefConfigSchema(z: Zod) {
+  return z.object({
+    kind: z.string(),
+    namespace: z.string(),
+    name: z.string(),
+  });
+}
+
+function getEntityGraphRelationsConfigSchema(z: Zod) {
+  // Mapping EntityRelationsGraphProps to config
+  // TODO: Define how className and render functions will be configured
+  return z.object({
+    rootEntityNames: getCompoundEntityRefConfigSchema(z)
+      .or(z.array(getCompoundEntityRefConfigSchema(z)))
+      .optional(),
+    kinds: z.array(z.string()).optional(),
+    relations: z.array(z.string()).optional(),
+    maxDepth: z.number().default(1),
+    unidirectional: z.boolean().optional(),
+    mergeRelations: z.boolean().optional(),
+    direction: z.nativeEnum(Direction).default(Direction.LEFT_RIGHT),
+    relationPairs: z
+      .array(z.tuple([z.string(), z.string()]))
+      .optional()
+      .default(ALL_RELATION_PAIRS),
+    zoom: z
+      .enum(['enabled', 'disabled', 'enable-on-click'])
+      .default('enable-on-click'),
+    curve: z
+      .enum(['curveStepBefore', 'curveMonotoneX'])
+      .default('curveMonotoneX'),
+  });
+}
+
+const CatalogGraphEntityCard = createEntityCardExtension({
+  name: 'entity-relations',
+  configSchema: createSchemaFromZod(z =>
+    z
+      .object({
+        // Filter is a config required to all entity cards
+        filter: z.string().optional(),
+        title: z.string().optional().default('Relations'),
+        height: z.number().optional(),
+        variant: z
+          .enum(['flex', 'fullHeight', 'gridItem'])
```

Comment:
Kinda wondering, should this apply at all in the new system? It's not entirely clear. I guess if one replaces the entity page entirely ... but hmm 🤔 

Kinda feeling that if this field should be a thing, then maybe it should be on `createEntityCardExtension` itself. Because if we want to keep this around, then it's gonna be a global concern for all cards everywhere.

Does it make sense to leave out for now, and set it to the value that makes the most sense in the current default grid, while we figure stuff out?

Reply:
Removed ✅